### ISSUE TITLE
how to decode cdx-base64 string?

### ISSUE THREAD

**Issue by biotech7** [epam/Indigo#846]
cdx base64 string is generated from ketcher, so how to decode this string to KET format string in Indigo? it's hard to find the api(JAVA Env.) to do this job.
could you help me to give an example?

**Comment by even1024** [comment]
Use "indigoLoadMoleculeFromString(String str)" and "Pointer indigoJson(int object);"

**Comment by biotech7** [comment]
In Java, could not figure out which api is equivalent to indigoLoadMoleculeFromString(String str).
the most similar api is loadStructure(String str,String params) . but how to set the "params", is there any manual?

**Comment by even1024** [comment]
Please check IndigoObject loadMolecule(String str)

**Comment by biotech7** [comment]
it returns an exception while trying to decode an base64 string. this api looks like treat base64 as SMILES.

```
Exception in thread "main" com.epam.indigo.IndigoException: molecule auto loader: SMILES loader: unrecognized lowercase symbol: m
	at com.epam.indigo.Indigo.checkResult(Indigo.java:105)
	at com.epam.indigo.Indigo.loadMolecule(Indigo.java:301)
	at test.test.test(test.java:33)
	at test.test.main(test.java:27)
```

**Comment by biotech7** [comment]
my cdx base64 code as follows, which is benzene

`VmpDRDAxMDAEAwIBAAAAAAAAAAAAAAAAAAAAAAMAEAAAAEFDRC9DaGVtU2tldGNoAYABAAAAEQgBAAESCAQANDNTAhMIBACE60kDA4ACAAAACgACAAEABIADAAAACgACAAIAAAIIAFK4XQDA+CUBAAQCAAEAAgQCAAYAKwQCAAEAAAAEgAQAAAAKAAIAAwAAAggA16NxAMD4JQEABAIAAQACBAIABgArBAIAAQAAAASABQAAAAoAAgAEAAACCACQwlMAFLgUAQAEAgABAAIEAgAGACsEAgABAAAABIAGAAAACgACAAUAAAIIAJqZewAUuBQBAAQCAAEAAgQCAAYAKwQCAAEAAAAEgAcAAAAKAAIABgAAAggAUrhdAOR3AwEABAIAAQACBAIABgArBAIAAQAAAASACAAAAAoAAgAHAAACCADXo3EA5HcDAQAEAgABAAIEAgAGACsEAgABAAAABYAJAAAACgACAAgABAYEAAQAAAAFBgQAAwAAAAAGAgABAAEGAgAAAAIGAgAAAAYGAQAAAQMCAAMABwgEAADAAAAAAAWACgAAAAoAAgAJAAQGBAAFAAAABQYEAAMAAAAABgIAAgABBgIAAAACBgIAAAAGBgEAAAQIAgCpAAEDAgADAAcIBAAAwAAAAAAFgAsAAAAKAAIACgAEBgQABgAAAAUGBAAEAAAAAAYCAAIAAQYCAAAAAgYCAAAABgYBAAAECAIAqQABAwIAAwAHCAQAAMAAAAAABYAMAAAACgACAAsABAYEAAcAAAAFBgQABQAAAAAGAgABAAEGAgAAAAIGAgAAAAYGAQAAAQMCAAMABwgEAADAAAAAAAWADQAAAAoAAgAMAAQGBAAIAAAABQYEAAYAAAAABgIAAQABBgIAAAACBgIAAAAGBgEAAAEDAgADAAcIBAAAwAAAAAAFgA4AAAAKAAIADQAEBgQACAAAAAUGBAAHAAAAAAYCAAIAAQYCAAAAAgYCAAAABgYBAAAECAIAqQABAwIAAwAHCAQAAMAAAAAAAAAAAAABBAABAAAAAAMOAAIA////////AAAAAAAAAAA=`

**Comment by even1024** [comment]
could you try to add "base64::" before your encoded string? Like:

base64::VmpDRDAxMDAEAwIBA.............etc

**Comment by biotech7** [comment]
cheers, it works!
but it will omit arrow symbol if decoding a cdx base64 reaction.

**Comment by even1024** [comment]
"but it will omit arrow symbol if decoding a cdx base64 reaction" - I am working on it. Will be fixed soon.

**Comment by biotech7** [comment]
so, supporting for  decoding  cdx base64 reagent/conditions reaction is highly preferred!

**Comment by even1024** [comment]
Implemented in 1.9,0.
